app/main.py

```diff
@@ -1,25 +1,76 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self._name = name
+
+    def __get__(self, instance: Any, owner: Any) -> Any:
+        if instance is None:
+            return self
+        return instance.__dict__.get(self._name)
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value must be an integer")
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError("Value out of range")
+        setattr(instance, self._name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self) -> None:
+        pass
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    weight = IntegerRange(20, 50)
+    height = IntegerRange(80, 120)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
 
 
 class Slide:
-    pass
+    def __init__(self,
+                 name: str,
+                 limitation_class:
+                 ChildrenSlideLimitationValidator
+                 | AdultSlideLimitationValidator
+                 ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+        limitation = self.limitation_class
+
+        try:
+            return (limitation.age.min_amount
+                    <= visitor.age
+                    <= limitation.age.max_amount
+                    and limitation.weight.min_amount
+                    <= visitor.weight
+                    <= limitation.weight.max_amount
+                    and limitation.height.min_amount
+                    <= visitor.height
+                    <= limitation.height.max_amount
+                    )
```

Comment:
The `can_access` method is currently checking the visitor's attributes against the `min_amount` and `max_amount` directly. Instead, it should rely on the descriptor's `__get__` method to access these attributes, which will automatically enforce the range checks.